mantidproject/mantid
# Added capability to generate offset files for ISIS Powder GEM automatically

Added capability to generate offset files for GEM based off of a run. This works in the same manner as the create_cal in Pearl.

**To test:**
run the following
```
from isis_powder import Gem
config_file_path = r"Path\to\Gem_config_example.yaml"
Gem_example = Gem(config_file=config_file_path, user_name="test")
Gem_example.create_cal(run_number=84258)
```
with these [files](https://github.com/mantidproject/mantid/files/2348797/Test.zip)
replacing the `config_file_path` with the path to the file within the zip, and update the paths within said file as well.

check your calibration directory, a folder for the cycle should have been generated, said folder should contain a .cal file with the offsets for the detectors. This test assumes you have access to the ISIS data search archive
Fixes #23241 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
